bowtie/_benchmarks.py

```diff
@@ -105,6 +105,32 @@ def get_benchmark_files(
     return files
 
 
+def merge_benchmark_report_jsons(
```

Comment:
You have too much of an affinity for dicts again it seems. Can't this be an operation on `Benchmark`s? Or `BenchmarkGroup`s, whichever is appropriate?

Reply:
I dont understand how we would do that, since to reduce time we are running each benchmark on separate jobs in the github workflow, it would create multiple reports which would be in json. So now when we want to merge them we have to load them in a dict right ? Thats why I did this

Reply:
Somewhere must have all the data, because somewhere you're running this code somewhere to do the merge -- so that place should be doing "load dicts into objects, call some method on that object which knows how to combine things correctly as Python objects" and then "write that result out to JSON", not "do arbitrary manipulation on dicts", essentially. But I admit I still don't have a mental picture of what the specific types are in that sentence, so let me know if I'm leading you astray.

Reply:
Okay so if I get you right:

In the workflow itself we can use the python shell to load the report into a `BenchmarkReport` using the from_dict method.

Now we simply call the `merge_benchmark_reports` function which takes objects as input and merges them to again produce an object.

We can then serialize this object and write it to json then.

Reply:
That sounds like it has the right shape at least, yeah. Possibly as `BenchmarkReport.merge(*paths).serialize()`

Reply:
> BenchmarkReport.merge(*paths).serializable()

BenchmarkReport.merge(*report_objects) right ?

Reply:
Yeah -- but some place has to be the place that actually *does* load from paths, so yeah I think that's right, at this layer take report objects, and then I guess in your `combine()` function you load paths into reports, yeah.

Reply:
But then if we have the combine function as well within bowtie itself then wouldnt it become the same thing as above. combine() + merge() would basically be doing the same thing, no?

Reply:
And here I am thinking, the combine and merge function would exist like this:

```
def combine_benchmark_reports(
    benchmark_report_files: list[Path],
) -> BenchmarkReport:
    return BenchmarkReport.merge([
        BenchmarkReport.from_dict(
            json.loads(report_file.read_text())
        )
        for report_file in benchmark_report_files
    ])
```
    
```
@classmethod
    def merge(
        cls, benchmark_reports: list[BenchmarkReport]
    ) -> BenchmarkReport:
        return BenchmarkReport(
            metadata=benchmark_reports[0].metadata,
            results={
                k: v for report in benchmark_reports
                for k, v in report.results.items()
            },
        )
```

Reply:
combine would call merge with the loaded paths, so it would just be your entry point, but that operation is now testable without file paths.

Reply:
That looks right -- though yeah presumably at some point there should be some validation that the metadata is mutually compatible, and not just arbitrarily take the first one.

Reply:
Thats true, but yeah then ig this looks right for an inital version.

Reply:
Yep.